ipynb/subsampleEMP.ipynb

```diff
@@ -0,0 +1,316 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "the next cell contains all parameters that might need to be changed"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "#filename of the input mapping file\n",
+    "file_mapping = ROOT + '/projects/emp/00-qiime-maps/merged/emp_qiime_mapping_latest.tsv'\n",
+    "#filename of the resulting mapping file\n",
+    "file_output = 'modMapping.txt'\n",
+    "\n",
+    "#define seed to make results comparable between runs\n",
+    "random.seed(7)\n",
+    "\n",
+    "#a sample must have at least 'minCounts' sequence reads to be included\n",
+    "minCounts = 1000 \n",
+    "\n",
+    "#sub-set sizes\n",
+    "setSizes = [2000, 10000, 20000]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "read in mapping file and filter accoring to two criteria:\n",
+    "1. a sample must contain a certain number of raw sequence reads\n",
+    "2. the sample must not be flagged as being a \"Control\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "metadata = pd.read_csv(file_mapping, sep=\"\\t\", index_col=0, low_memory=False)\n",
+    "filtered_metadata = metadata[\n",
+    "    (metadata['sequences_per_sample'] >= minCounts) & #remove samples with low read numbers\n",
+    "    (metadata['empo_1'] != 'Control')                 #remove controls \n",
+    "]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## actual logic to pick samples\n",
+    "Assume that we want to create more than one subsample set. However, they should form a strict hierarchy, i.e. a sample in a smaller set must also occure in any larger sets.\n",
+    "\n",
+    "Furthermore, we want to make sure that each group is covered, since some groups are very large, others are small and might be otherwise subject of being missed. We divide the number of targeted samples by the number of existing groups (+ some corrections to end up with the right number of samples) and randomly sample within each group a sufficient number of samples.\n",
+    "\n",
+    "If a group has more samples than needed samples should drawn from all existing studies in that group.\n",
+    "\n",
+    "A larger subset is always initialized with the IDs of the next smaller subset, thus we make sure to form the hierarchy."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "#convert np.infty to the actual number of total available samples in EMP satisfying the filtering criteria\n",
+    "for i in range(0, len(setSizes)): \n",
+    "    if setSizes[i] is np.infty:\n",
+    "        setSizes[i] = filtered_metadata.shape[0]\n",
+    "        \n",
```

Comment:
Just in case you were curious, you can get rid of this loop doing the following
```python
f = lambda x: filtered_metadata.shape[0] if x is np.infty else x
map(f, setSizes)
```

More of a functional approach - figured you'd prefer this.